### ISSUE TITLE
Unable to Stop Jobs

### ISSUE THREAD

**Issue by moraneden** [hashicorp/nomad#9722]
### Nomad version
Nomad v1.0.1 (c9c68aa55a7275f22d2338f2df53e67ebfcb9238)

### Operating system and Environment details

windows server 2016 version 1607

### Issue
We ran into a scenario (it happens multiple times), when trying to stop a job, 
Nomad doesn't stop current allocations and add more allocation when the job start again.
* Run nomad gc, doesn't help.
* Killing the process doesnt help
* Only restart nomad service solves the issue.

![image](https://user-images.githubusercontent.com/15603334/103645497-bd850c00-4f60-11eb-8a73-79eee2eeb87e.png)

### Reproduction steps

define nomad client with   gc_max_allocs = 1000
start job - 
driver = "raw_exec"
3 processes

just stop and start until the processes wont stop anymore.

### Job file (if appropriate)
```
job "Test" {
  datacenters = ["dc1"]
  type = "service" 
  spread {
	attribute = "${node.unique.name}"
	weight    = 100
  } 
  constraint {
    attribute = "${attr.kernel.name}"
    value     = "windows"
  }
  #
  group "LabMD-test" {
    count = 3
    restart {
      attempts = 10
      interval = "3m"
      delay = "15s"
      mode = "delay"
    }
    
    task "Labmd-test" {

      driver = "raw_exec"
	  kill_timeout = "30s"
		  
	  resources {
        network {
          port "laas_port" {}
        }
	  }
	  service {
				name = "LabMDServer-test"
				tags = ["urlprefix-/LabMDServert strip=/LabMDServert"]
				port = "laas_port"			
				check {
					type     = "http"
					port     = "laas_port"
					path     = "/status"
					interval = "30s"
					timeout  = "10s"
				}			
			    check_restart {
				  limit = 2
				  grace = "90s"
				  ignore_warnings = false
				}
	  }
      config {
        command = "SOMEXE.exe"
      }
    }
  }
}
```
### Nomad Client logs (if appropriate)

Attach log when trying to stop and start the job (Labmd-Service-heb-test)

[1.txt](https://github.com/hashicorp/nomad/files/5770070/1.txt)

**Comment by notnoop** [comment]
@moraneden That's very perplexing indeed, and thanks for reaching out.  We'll investigate.

To ease debugging, we would love to get the goroutine stack traces, via the [pprof api](https://www.nomadproject.io/api-docs/agent#agent-runtime-profiles), e.g. `curl https://localhost:4646/v1/agent/pprof/goroutine`.  Seeing the state of the agent will help us narrow down the problem earlier, while we attempt to reproduce it.

**Comment by moraneden** [comment]
@notnoop Hi,
We dont use https, 
I tried curl http://localhost:4646/v1/agent/pprof/goroutine 
and got "Permission denied"

**Comment by notnoop** [comment]
You may need to set the authentication token if ACL is enabled, and/or set enable_debug flag: https://www.nomadproject.io/docs/configuration#enable_debug .  Ideally, you'd collect the stacktrace on the client when an alloc gets into the unstoppable state.  Thanks!

**Comment by moraneden** [comment]
@notnoop  I have the file you requested, where can I send it to?

**Comment by notnoop** [comment]
Great!  You can post here, it shouldn't contain any sensitive info.  Alternatively, you can also email nomad-oss-debug@hashicorp.com .

**Comment by moraneden** [comment]
@notnoop I sent it yesterday to nomad-oss-debug@hashicorp.com 
Did you get it?

**Comment by notnoop** [comment]
Sorry for the slow response.  Yes, I have it and plan to investigate further.  Will post an update when I get any leads.  Thanks!

**Comment by moraneden** [comment]
@notnoop  any new?

**Comment by moraneden** [comment]
@notnoop any update?

**Comment by tgross** [comment]
I took a quick look through the goroutine profile you sent. Weirdly it looks like most of the goroutines are parked (that is, not o CPU):

```
(pprof) top
Showing nodes accounting for 3711, 99.95% of 3713 total
Dropped 232 nodes (cum <= 18)
Showing top 10 nodes out of 120
      flat  flat%   sum%        cum   cum%
      3452 92.97% 92.97%       3452 92.97%  runtime.gopark
       259  6.98% 99.95%        259  6.98%  runtime.cgocall
         0     0% 99.95%        130  3.50%  bufio.(*Reader).Read
         0     0% 99.95%        128  3.45%  bufio.(*Reader).ReadLine
         0     0% 99.95%        128  3.45%  bufio.(*Reader).ReadSlice
         0     0% 99.95%        141  3.80%  bufio.(*Reader).fill
         0     0% 99.95%        125  3.37%  bufio.(*Scanner).Scan
         0     0% 99.95%         98  2.64%  context.propagateCancel.func1
         0     0% 99.95%        399 10.75%  github.com/LK4D4/joincontext.(*joinContext).run
         0     0% 99.95%         29  0.78%  github.com/hashicorp/consul/api.(*Agent).Services
```

I dumped this to a graphical format:

[pprof001.svg.zip](https://github.com/hashicorp/nomad/files/5876090/pprof001.svg.zip)

And after looking at that, I noticed a large number of deploymentwatcher goroutines:

```
(pprof) top show deployment
Active filters:
   focus=show|deployment
Showing nodes accounting for 602, 16.21% of 3713 total
Dropped 3 nodes (cum <= 18)
      flat  flat%   sum%        cum   cum%
       602 16.21% 16.21%        602 16.21%  runtime.gopark
         0     0% 16.21%        579 15.59%  github.com/hashicorp/go-memdb.WatchSet.WatchCtx
         0     0% 16.21%        579 15.59%  github.com/hashicorp/go-memdb.watchFew
         0     0% 16.21%        578 15.57%  github.com/hashicorp/nomad/nomad/deploymentwatcher.(*deploymentWatcher).getAllocs
         0     0% 16.21%        578 15.57%  github.com/hashicorp/nomad/nomad/deploymentwatcher.(*deploymentWatcher).getAllocsCh.func1
         0     0% 16.21%         22  0.59%  github.com/hashicorp/nomad/nomad/deploymentwatcher.(*deploymentWatcher).watch
         0     0% 16.21%        579 15.59%  github.com/hashicorp/nomad/nomad/state.(*StateStore).BlockingQuery
         0     0% 16.21%        602 16.21%  runtime.selectgo
```

So it looks like the servers are creating a whole lot of deployments as you say, but they don't seem to be marked complete.

Just for clarity, when you say the job is stopping, do you mean:

1. The allocations are marked `complete` but the actual `SOMEXE.exe` processes are continuing?
2. The allocations are never marked `complete`?

It might help to get the output of `nomad job status Test` for a job that's in this state, as well as the output of `nomad deployment list` and `nomad deployment status :deployment_id` for the running deployment.

**Comment by moraneden** [comment]
@tgross  thank you! 
please see  below - 
nomad job status:

ID            = Laas-Service
Name          = Laas-Service
Submit Date   = 2021-01-28T09:19:34+02:00
Type          = service
Priority      = 50
Datacenters   = dc1
Namespace     = default
Status        = running
Periodic      = false
Parameterized = false

Summary
Task Group  Queued  Starting  Running  Failed  Complete  Lost
Laas        0       0         5        0       102       0

Latest Deployment
ID          = 4022ec22
Status      = failed
Description = Failed due to progress deadline

Deployed
Task Group  Desired  Placed  Healthy  Unhealthy  Progress Deadline
Laas        2        2       1        1          2021-01-28T09:29:47+02:00

Allocations
ID        Node ID   Task Group  Version  Desired  Status    Created     Modified
5eaf2534  2d992ec2  Laas        104      run      running   49m49s ago  44m49s ago
a51b7bf2  9240830d  Laas        104      run      running   49m49s ago  49m35s ago
130be345  9240830d  Laas        102      stop     complete  4h3m ago    50m4s ago
6f79fab7  2d992ec2  Laas        102      stop     running   4h3m ago    55m6s ago
c7bc6eef  2d992ec2  Laas        100      stop     running   5h31m ago   4h5m ago
0517f1b2  2d992ec2  Laas        98       stop     running   15h20m ago  5h36m ago


nomad deployment status 4022ec22
ID          = 4022ec22
Job ID      = Laas-Service
Job Version = 104
Status      = failed
Description = Failed due to progress deadline

Deployed
Task Group  Desired  Placed  Healthy  Unhealthy  Progress Deadline
Laas        2        2       1        1          2021-01-28T09:29:47+02:00

**Comment by tgross** [comment]
Ok, so the allocation statuses for that job looks like this:

```
Allocations
ID        Node ID   Task Group  Version  Desired  Status    Created     Modified
5eaf2534  2d992ec2  Laas        104      run      running   49m49s ago  44m49s ago
a51b7bf2  9240830d  Laas        104      run      running   49m49s ago  49m35s ago
130be345  9240830d  Laas        102      stop     complete  4h3m ago    50m4s ago
6f79fab7  2d992ec2  Laas        102      stop     running   4h3m ago    55m6s ago
c7bc6eef  2d992ec2  Laas        100      stop     running   5h31m ago   4h5m ago
0517f1b2  2d992ec2  Laas        98       stop     running   15h20m ago  5h36m ago
```

We see that we've got 1 straggler from 3 different versions of the job, all of which are on the same node. I note that you have `spread` by node ID, so I'm not sure if the job failure you're seeing on the most recent deploy is because the task needs a resource that's in use by the task that won't stop.

But if I extract the relevant log lines from the ones you posted for the `nomad job stop` I see a few things:

* We see that the server is definitely telling the clients to stop the job as we'd expect.
* We see the update message hitting the client, telling it to update the alloc status.
* We don't see any logs from the client for the shutdown activity.
* These appear to be combined server-client nodes, which is not a recommended production configuration. It _should_ be ok if you don't have heavy workloads but there might be weird gotchas.

<details><summary>relevant log lines</summary>

```
2021-01-05T14:03:18.157+0200 [DEBUG] http: request complete: method=DELETE path=/v1/job/Labmd-Service-heb-test duration=20.9907ms
...
2021-01-05T14:03:18.160+0200 [DEBUG] worker: dequeued evaluation: eval_id=9e14aff4-847b-74be-ef71-516ab58ad324
2021-01-05T14:03:18.161+0200 [DEBUG] worker.service_sched: reconciled current state with desired state: eval_id=9e14aff4-847b-74be-ef71-516ab58ad324 job_id=Labmd-Service-heb-test namespace=default results="Total changes: (place 0) (destructive 0) (inplace 0) (stop 3)
Desired Changes for "LabMD-test": (place 0) (inplace 0) (destructive 0) (stop 3) (migrate 0) (ignore 0) (canary 0)"
2021-01-05T14:03:18.177+0200 [DEBUG] worker: submitted plan for evaluation: eval_id=9e14aff4-847b-74be-ef71-516ab58ad324
2021-01-05T14:03:18.177+0200 [DEBUG] worker.service_sched: setting eval status: eval_id=9e14aff4-847b-74be-ef71-516ab58ad324 job_id=Labmd-Service-heb-test namespace=default status=complete
...
2021-01-05T14:03:18.207+0200 [DEBUG] worker: updated evaluation: eval="<Eval "9e14aff4-847b-74be-ef71-516ab58ad324" JobID: "Labmd-Service-heb-test" Namespace: "default">"
2021-01-05T14:03:18.207+0200 [DEBUG] worker: ack evaluation: eval_id=9e14aff4-847b-74be-ef71-516ab58ad324
...
2021-01-05T14:03:18.212+0200 [DEBUG] client: updated allocations: index=6891263 total=153 pulled=3 filtered=150
2021-01-05T14:03:18.212+0200 [DEBUG] client: allocation updates: added=0 removed=0 updated=3 ignored=150
...
2021-01-05T14:03:18.410+0200 [DEBUG] client: updated allocations: index=6891265 total=153 pulled=0 filtered=153
2021-01-05T14:03:18.410+0200 [DEBUG] client: allocation updates: added=0 removed=0 updated=0 ignored=153
2021-01-05T14:03:18.410+0200 [DEBUG] client: allocation updates applied: added=0 removed=0 updated=0 ignored=153 errors=0
```

</details>

My hypothesis here is that the workload is not responding to the signal that's being sent or the client is somehow hanging during the shutdown process. I would be very interested to see client logs for a node where a task won't stop (like `2d992ec2`) around the time the job is stopped.

**Comment by moraneden** [comment]
[out.log](https://github.com/hashicorp/nomad/files/5924203/out.log)

@tgross  I attached the debug log,
this time it was  task_name=HtmlToPDF
I see some errors around - 2021-02-03T03:01:14.684+0200

 2021-02-03T03:01:14.684+0200 [ERROR] client.alloc_runner.task_runner: error recovering task; cleaning up: alloc_id=ea19c903-4fbd-9ba2-dbb4-0deace52eff1 task=HtmlToPDF error="failed to reattach to executor: error creating rpc client for executor plugin: Reattachment process not found" task_id=ea19c903-4fbd-9ba2-dbb4-0deace52eff1/HtmlToPDF/b45e1e6e

nomad job status HtmlToPDF
ID            = HtmlToPDF
Name          = HtmlToPDF
Submit Date   = 2021-02-04T10:14:21+02:00
Type          = service
Priority      = 50
Datacenters   = dc1
Namespace     = default
Status        = dead (stopped)
Periodic      = false
Parameterized = false

Summary
Task Group  Queued  Starting  Running  Failed  Complete  Lost
HtmlToPDF   0       0         6        0       29        2

Latest Deployment
ID          = b7dcd5f3
Status      = cancelled
Description = Cancelled because job is stopped

Deployed
Task Group  Desired  Placed  Healthy  Unhealthy  Progress Deadline
HtmlToPDF   2        2       1        1          2021-02-04T10:24:45+02:00

Allocations
ID        Node ID   Task Group  Version  Desired  Status    Created     Modified
4120faab  9240830d  HtmlToPDF   96       stop     complete  14m45s ago  10m22s ago
991df09a  2d992ec2  HtmlToPDF   96       stop     running   14m45s ago  9m45s ago
52e4b211  9240830d  HtmlToPDF   94       stop     complete  4h23m ago   14m57s ago
c476e61a  2d992ec2  HtmlToPDF   94       stop     running   4h23m ago   20m51s ago
ee1dba8b  2d992ec2  HtmlToPDF   92       stop     running   19h15m ago  4h25m ago
9f218cf2  2d992ec2  HtmlToPDF   90       stop     running   22h52m ago  19h16m ago
c8ba3751  2d992ec2  HtmlToPDF   88       stop     running   1d4h ago    22h53m ago
29f8c800  2d992ec2  HtmlToPDF   86       stop     running   1d6h ago    1d4h ago

**Comment by tgross** [comment]
Thanks @moraneden. One thing that immediately jumps out at me here is that this combined server-client node has restarted with running tasks. So when the client comes back up, it hasn't heartbeat to the servers because its "own" server has also been down. That server may have a stale view of the world either because the other servers may have gone on without it (if they have quorum). And then it looks like some of the processes that the client expects to see when it's restarted are actually gone, which is why you're getting all the "error recovering task" errors. That puts the client in a race with the server as it tries to restart those tasks and the server is likely telling the client those allocations are "lost" and shouldn't be restarted, so they can be rescheduled on other hosts. I also see "found an alloc without any local state" warnings, which suggests that you might have removed the client state store?

Overall, just a bad situation and that's why we document that you shouldn't be running combined server-client instances. If you are going to do that, I would strongly recommend that you `nomad node drain` the host before shutting it down. At the very least, that will make it the behavior a lot more understandable when it restarts so that we can decipher what's going on with these processes.

All that being said, to start to debug what's happening, let's go through the logs and follow the lifecycle of a single allocation. If we look at the job status output, we can see the allocation `991df09a` looks dubious:

```
ID       Node ID  Task Group  Version  Desired  Status    Created     Modified
4120faab 9240830d HtmlToPDF   96       stop     complete  14m45s ago  10m22s ago
991df09a 2d992ec2 HtmlToPDF   96       stop     running   14m45s ago  9m45s ago
```

Let's start with when we submitted that job. We see the post that results in 2 placements, and one of the placements landing on the node we're watching the logs on (I've elided log lines belonging to other jobs or allocs with `...` in my notes below):

```
    2021-02-04T10:14:21.041+0200 [DEBUG] http: request complete: method=POST path=/v1/job/HtmlToPDF duration=15.0004ms
    2021-02-04T10:14:21.041+0200 [DEBUG] worker: dequeued evaluation: eval_id=c4757621-5d2f-4f6b-3bf5-0d47a0485686
    2021-02-04T10:14:21.041+0200 [DEBUG] worker.service_sched: reconciled current state with desired state: eval_id=c4757621-5d2f-4f6b-3bf5-0d47a0485686 job_id=HtmlToPDF namespace=default results="Total changes: (place 2) (destructive 0) (inplace 0) (stop 0)
Created Deployment: "b7dcd5f3-a4d2-b696-80e0-fa9490e85b7e"
Desired Changes for "HtmlToPDF": (place 2) (inplace 0) (destructive 0) (stop 0) (migrate 0) (ignore 0) (canary 0)"
    2021-02-04T10:14:21.054+0200 [DEBUG] worker: submitted plan for evaluation: eval_id=c4757621-5d2f-4f6b-3bf5-0d47a0485686
    2021-02-04T10:14:21.055+0200 [DEBUG] worker.service_sched: setting eval status: eval_id=c4757621-5d2f-4f6b-3bf5-0d47a0485686 job_id=HtmlToPDF namespace=default status=complete
    2021-02-04T10:14:21.057+0200 [DEBUG] client: updated allocations: index=7667565 total=47 pulled=2 filtered=45
    2021-02-04T10:14:21.057+0200 [DEBUG] client: allocation updates: added=1 removed=0 updated=1 ignored=45
...
    2021-02-04T10:14:21.067+0200 [DEBUG] worker: updated evaluation: eval="<Eval "c4757621-5d2f-4f6b-3bf5-0d47a0485686" JobID: "HtmlToPDF" Namespace: "default">"
    2021-02-04T10:14:21.067+0200 [DEBUG] worker: ack evaluation: eval_id=c4757621-5d2f-4f6b-3bf5-0d47a0485686
    2021-02-04T10:14:21.072+0200 [DEBUG] client: allocation updates applied: added=1 removed=0 updated=1 ignored=45 errors=0
    2021-02-04T10:14:21.074+0200 [DEBUG] client.alloc_runner.task_runner: lifecycle start condition has been met, proceeding: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 task=HtmlToPDF
    2021-02-04T10:14:21.095+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon: starting plugin: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 task=HtmlToPDF path=C:\Tools\Nomad\nomad.exe args=[C:\Tools\Nomad\nomad.exe, logmon]
    2021-02-04T10:14:21.097+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon: plugin started: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 task=HtmlToPDF path=C:\Tools\Nomad\nomad.exe pid=20176
    2021-02-04T10:14:21.097+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon: waiting for RPC address: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 task=HtmlToPDF path=C:\Tools\Nomad\nomad.exe
    2021-02-04T10:14:21.140+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon: using plugin: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 task=HtmlToPDF version=2
    2021-02-04T10:14:21.140+0200 [DEBUG] client.alloc_runner.task_runner.task_hook.logmon.nomad.exe: plugin address: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 task=HtmlToPDF @module=logmon address=127.0.0.1:10029 network=tcp timestamp=2021-02-04T10:14:21.118+0200
    2021-02-04T10:14:21.142+0200 [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad.exe: opening fifo: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 task=HtmlToPDF path=//./pipe/HtmlToPDF-f13b656a.stdout @module=logmon timestamp=2021-02-04T10:14:21.142+0200
    2021-02-04T10:14:21.143+0200 [INFO]  client.alloc_runner.task_runner.task_hook.logmon.nomad.exe: opening fifo: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 task=HtmlToPDF @module=logmon path=//./pipe/HtmlToPDF-f13b656a.stderr timestamp=2021-02-04T10:14:21.143+0200
    2021-02-04T10:14:21.187+0200 [WARN]  client.alloc_runner.task_runner: some environment variables not available for rendering: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 task=HtmlToPDF keys=
    2021-02-04T10:14:21.187+0200 [DEBUG] client.driver_mgr.raw_exec.executor: starting plugin: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 driver=raw_exec task_name=HtmlToPDF path=C:\Tools\Nomad\nomad.exe args=[C:\Tools\Nomad\nomad.exe, executor, {"LogFile":"C:\\Tools\\Nomad\\Data\\alloc\\991df09a-644c-92e7-13bd-e99904bf3dc3\\HtmlToPDF\\executor.out","LogLevel":"debug","FSIsolation":false}]
    2021-02-04T10:14:21.189+0200 [DEBUG] client.driver_mgr.raw_exec.executor: plugin started: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 driver=raw_exec task_name=HtmlToPDF path=C:\Tools\Nomad\nomad.exe pid=11200
    2021-02-04T10:14:21.189+0200 [DEBUG] client.driver_mgr.raw_exec.executor: waiting for RPC address: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 driver=raw_exec task_name=HtmlToPDF path=C:\Tools\Nomad\nomad.exe
...
    2021-02-04T10:14:22.288+0200 [DEBUG] client.driver_mgr.raw_exec.executor: using plugin: alloc_id=991df09a-644c-92e7-13bd-e99904bf3dc3 driver=raw_exec task_name=HtmlToPDF version=2
...
    2021-02-04T10:14:48.058+0200 [DEBUG] worker.service_sched: reconciled current state with desired state: eval_id=c48e4196-d554-6d7b-6d2b-f5634a4fabde job_id=HtmlToPDF namespace=default results="Total changes: (place 0) (destructive 0) (inplace 0) (stop 0)
Desired Changes for "HtmlToPDF": (place 0) (inplace 0) (destructive 0) (stop 0) (migrate 0) (ignore 2) (canary 0)"
    2021-02-04T10:14:48.058+0200 [DEBUG] worker.service_sched: setting eval status: eval_id=c48e4196-d554-6d7b-6d2b-f5634a4fabde job_id=HtmlToPDF namespace=default status=complete
    2021-02-04T10:14:48.069+0200 [DEBUG] worker: updated evaluation: eval="<Eval "c48e4196-d554-6d7b-6d2b-f5634a4fabde" JobID: "HtmlToPDF" Namespace: "default">"
```

So far so good. We've got a running allocation. Then we see an API call to stop the job only 4 minutes later. That's not what I'd expect for a `service` job but hey, it's your workload. :grinning: The scheduler issues 2 "stop" and the evaluation is acknowledged by the planner. We see the client gets two updates for the allocation, which should be it telling it to stop. One of the updates the client is getting appears to be for a different job (alloc `6100a643`) but is being discarded, which only happens if it has a stale update that it's since received a newer one. The note about garbage collection being skipped is just saying we're over the GC threshold but there aren't any stopped allocations to cleanup.

```
    2021-02-04T10:18:43.422+0200 [DEBUG] http: request complete: method=GET path=/v1/job/HtmlToPDF/evaluations?index=7667959 duration=9.428021s
    2021-02-04T10:18:43.422+0200 [DEBUG] worker: dequeued evaluation: eval_id=9aebbecf-b43a-1742-83d9-f736b7b1f397
    2021-02-04T10:18:43.422+0200 [DEBUG] http: request complete: method=GET path=/v1/job/HtmlToPDF?index=7667565 duration=9.4180181s
    2021-02-04T10:18:43.423+0200 [DEBUG] worker.service_sched: reconciled current state with desired state: eval_id=9aebbecf-b43a-1742-83d9-f736b7b1f397 job_id=HtmlToPDF namespace=default results="Total changes: (place 0) (destructive 0) (inplace 0) (stop 2)
Deployment Update for ID "b7dcd5f3-a4d2-b696-80e0-fa9490e85b7e": Status "cancelled"; Description "Cancelled because job is stopped"
Desired Changes for "HtmlToPDF": (place 0) (inplace 0) (destructive 0) (stop 2) (migrate 0) (ignore 0) (canary 0)"
    2021-02-04T10:18:43.422+0200 [DEBUG] http: request complete: method=DELETE path=/v1/job/HtmlToPDF duration=82.9969ms
    2021-02-04T10:18:43.443+0200 [DEBUG] worker: submitted plan for evaluation: eval_id=9aebbecf-b43a-1742-83d9-f736b7b1f397
    2021-02-04T10:18:43.443+0200 [DEBUG] worker.service_sched: setting eval status: eval_id=9aebbecf-b43a-1742-83d9-f736b7b1f397 job_id=HtmlToPDF namespace=default status=complete
    2021-02-04T10:18:43.444+0200 [DEBUG] http: request complete: method=GET path=/v1/job/HtmlToPDF/allocations?index=7667997 duration=9.4510304s
    2021-02-04T10:18:43.447+0200 [DEBUG] http: request complete: method=GET path=/v1/job/HtmlToPDF/deployment?index=7667958 duration=7.4500031s
    2021-02-04T10:18:43.470+0200 [DEBUG] http: request complete: method=GET path=/v1/job/HtmlToPDF?index=7668001 duration=0s
    2021-02-04T10:18:43.475+0200 [DEBUG] client: updated allocations: index=7668002 total=56 pulled=2 filtered=54
    2021-02-04T10:18:43.475+0200 [DEBUG] client: allocation updates: added=0 removed=0 updated=2 ignored=54
    2021-02-04T10:18:43.490+0200 [DEBUG] client.alloc_runner: Discarding allocation update: alloc_id=6100a643-e5fc-0825-f010-8d17200a6bc9 skipped_modify_index=7663488 new_modify_index=7663488
    2021-02-04T10:18:43.492+0200 [DEBUG] worker: updated evaluation: eval="<Eval "9aebbecf-b43a-1742-83d9-f736b7b1f397" JobID: "HtmlToPDF" Namespace: "default">"
    2021-02-04T10:18:43.492+0200 [DEBUG] worker: ack evaluation: eval_id=9aebbecf-b43a-1742-83d9-f736b7b1f397
    2021-02-04T10:18:43.504+0200 [DEBUG] client: allocation updates applied: added=0 removed=0 updated=2 ignored=54 errors=0
    2021-02-04T10:18:43.511+0200 [DEBUG] http: request complete: method=GET path=/v1/job/HtmlToPDF/evaluations?index=7668001 duration=6.0007ms
    2021-02-04T10:18:43.512+0200 [INFO]  client.gc: garbage collection skipped because no terminal allocations: reason="number of allocations (53) is over the limit (50)"
    2021-02-04T10:18:43.597+0200 [DEBUG] http: request complete: method=GET path=/v1/job/HtmlToPDF/allocations?index=7668002 duration=55.0038ms
    2021-02-04T10:18:43.598+0200 [DEBUG] client: updated allocations: index=7668005 total=56 pulled=1 filtered=55
    2021-02-04T10:18:43.598+0200 [DEBUG] client: allocation updates: added=0 removed=0 updated=1 ignored=55
    2021-02-04T10:18:43.606+0200 [DEBUG] client.alloc_runner: Discarding allocation update: alloc_id=6100a643-e5fc-0825-f010-8d17200a6bc9 skipped_modify_index=7663488 new_modify_index=7663488
    2021-02-04T10:18:43.606+0200 [DEBUG] client: allocation updates applied: added=0 removed=0 updated=1 ignored=55 errors=0
    2021-02-04T10:18:43.606+0200 [INFO]  client.gc: garbage collection skipped because no terminal allocations: reason="number of allocations (53) is over the limit (50)"
    2021-02-04T10:18:43.649+0200 [DEBUG] http: request complete: method=GET path=/v1/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3 duration=996.6µs
    2021-02-04T10:18:43.649+0200 [DEBUG] http: request complete: method=GET path=/v1/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3 duration=996.6µs
```

And then from then on we just see queries from the CLI or web UI for the allocation. Nothing happens.

<details><summary>uninteresting logs</summary>

```
    2021-02-04T10:18:43.790+0200 [DEBUG] http: request complete: method=GET path=/v1/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3 duration=1.0023ms
    2021-02-04T10:18:43.847+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:18:45.596+0200 [DEBUG] http: request complete: method=GET path=/v1/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3 duration=0s
    2021-02-04T10:18:45.624+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:18:48.150+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:18:50.662+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=898.6µs
    2021-02-04T10:18:53.172+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.0039ms
    2021-02-04T10:18:55.680+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=984.7µs
    2021-02-04T10:18:58.188+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:00.697+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:03.209+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:05.718+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.1473ms
    2021-02-04T10:19:08.225+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:10.734+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:13.243+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:15.751+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=997.4µs
    2021-02-04T10:19:18.259+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:20.770+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:21.429+0200 [DEBUG] http: request complete: method=GET path=/v1/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3 duration=0s
    2021-02-04T10:19:21.451+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:23.993+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=996.5µs
    2021-02-04T10:19:26.503+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.0662ms
    2021-02-04T10:19:29.014+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=997.8µs
    2021-02-04T10:19:31.523+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:34.035+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.9939ms
    2021-02-04T10:19:36.547+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:39.061+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:41.582+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=6.5278ms
    2021-02-04T10:19:44.115+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:46.631+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:49.140+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:51.651+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:54.159+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:56.665+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:19:59.174+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:01.699+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=747.1µs
    2021-02-04T10:20:04.207+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:06.714+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:09.237+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=902µs
    2021-02-04T10:20:11.748+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:14.263+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:16.777+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.0019ms
    2021-02-04T10:20:19.287+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:21.797+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:24.306+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.0035ms
    2021-02-04T10:20:26.814+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:29.328+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:32.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:36.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:40.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:44.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:48.380+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.3754ms
    2021-02-04T10:20:52.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:20:56.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:21:00.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:21:04.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:21:08.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=996.6µs
    2021-02-04T10:21:12.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.1707ms
    2021-02-04T10:21:16.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:21:20.380+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=92.6µs
    2021-02-04T10:21:24.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:21:28.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:21:32.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:21:36.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:21:40.390+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=957.3µs
    2021-02-04T10:21:44.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:21:48.380+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=2.977ms
    2021-02-04T10:21:52.379+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:21:56.375+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:00.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:04.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:08.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:12.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:16.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:20.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:24.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=997.4µs
    2021-02-04T10:22:28.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:32.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:36.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.1354ms
    2021-02-04T10:22:40.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:44.375+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:48.379+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.0019ms
    2021-02-04T10:22:52.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:22:56.385+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=999µs
    2021-02-04T10:23:00.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:04.149+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:07.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.0162ms
    2021-02-04T10:23:11.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:15.382+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:19.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:23.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:27.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:31.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:35.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=2.0013ms
    2021-02-04T10:23:39.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:43.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:47.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:51.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:23:55.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=906.5µs
    2021-02-04T10:23:59.380+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:24:03.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:24:07.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:24:11.383+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:24:15.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=263.3µs
    2021-02-04T10:24:19.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:24:23.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=999.8µs
    2021-02-04T10:24:27.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:24:31.392+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:24:35.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:24:39.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:24:43.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:24:47.378+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.1903ms
    2021-02-04T10:24:51.380+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=990.5µs
    2021-02-04T10:24:55.379+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=296.1µs
    2021-02-04T10:24:59.376+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:25:03.382+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.0109ms
    2021-02-04T10:25:07.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=997.4µs
    2021-02-04T10:25:11.388+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=4.8948ms
    2021-02-04T10:25:15.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:25:19.377+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:25:21.902+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=0s
    2021-02-04T10:25:24.420+0200 [DEBUG] http: request complete: method=GET path=/v1/client/allocation/991df09a-644c-92e7-13bd-e99904bf3dc3/stats duration=1.0195ms
```

</details>

So it looks like the client is getting the stop but we never see anything happen with that. This suggests the shutdown is being blocked. Trace-level logs would tell us if its being blocked by one of the allocation runner hooks, but it's possible there's a problem with the signals we're sending to the executor to stop the process. Does your workload mask signals or possibly reload it's configuration on signals?

**Comment by moraneden** [comment]
@tgross  the issue here is that most of the time its works fine, meaning, the nomad sends stop signal and the process (workload) going down nicely,
sometimes the issue above happens, 
moreover, when the issue happen, I expect nomad the kill the workload if it doesn't go down nicely, right?

**Comment by tgross** [comment]
> when the issue happen, I expect nomad the kill the workload if it doesn't go down nicely, right?

Absolutely! That's why I was asking if the application masks signals... I'm trying to figure out if our [Windows executor](https://github.com/hashicorp/nomad/blob/master/drivers/shared/executor/executor_windows.go) might be sending the wrong signals for shutdown here.

**Comment by moraneden** [comment]
@tgross thank you for your help! 
let me know if you need anything from me.

**Comment by moraneden** [comment]
@tgross  quick update - I moved all my processes to windows 2019 (from win 2016) and since than I didn't experience the issue...

**Comment by mikenomitch** [comment]
I'm going to close this issue out since moving to Windows 2019 has stopped this issue. If anybody is on this version or later and sees this issue, let us know and we can reopen it!

**Comment by Yatekii** [comment]
I have exactly this issue running nomad client/server on ubuntu and controlling it via macOS CLI :/
